src/canopy/knowledge_base/record_encoder/octoai.py

```diff
@@ -0,0 +1,68 @@
+import os
+from typing import List
+from pinecone_text.dense.openai_encoder import OpenAIEncoder
+from canopy.knowledge_base.models import KBDocChunk, KBEncodedDocChunk, KBQuery
+from canopy.knowledge_base.record_encoder.dense import DenseRecordEncoder
+from canopy.models.data_models import Query
+
+OCTOAI_BASE_URL = "https://text.octoai.run/v1"
+
+
+class OctoAIRecordEncoder(DenseRecordEncoder):
+    """
+    OctoAIRecordEncoder is a type of DenseRecordEncoder that uses the OpenAI `embeddings` API.
+    The implementation uses the `OpenAIEncoder` class from the `pinecone-text` library.
+    For more information about see: https://github.com/pinecone-io/pinecone-text
+
+    """  # noqa: E501
+    """
+    Initialize the OctoAIRecordEncoder
+
+    Args:
+        api_key: The OctoAI Endpoint API Key
+        base_url: The Base URL for the OctoAI Endpoint
+        model_name: The name of the OctoAI embeddings model to use for encoding. See https://octo.ai/docs/text-gen-solution/getting-started
+        batch_size: The number of documents or queries to encode at once.
+                    Defaults to 1.
+        **kwargs: Additional arguments to pass to the underlying `pinecone-text. OpenAIEncoder`.
+    """  # noqa: E501
+    def __init__(self,
+                 *,
+                 api_key: str = "",
+                 base_url: str = OCTOAI_BASE_URL,
+                 model_name: str = "thenlper/gte-large",
+                 batch_size: int = 1024,
+                 **kwargs):
+
+        ae_api_key = api_key or os.environ.get("OCTOAI_API_KEY")
```

Comment:
ae stands for anyscale so maybe worth changing that to octoai_api_key (same for ae_base_url below)